azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSFileIO.java

```diff
@@ -126,6 +131,8 @@ public void initialize(Map<String, String> props) {
     this.properties = SerializableMap.copyOf(props);
     this.azureProperties = new AzureProperties(properties);
     initMetrics(properties);
+    this.vendedAdlsCredentialProvider =
+        this.azureProperties.vendedAdlsCredentialProvider().orElse(null);
```

Comment:
```suggestion
    this.azureProperties.vendedAdlsCredentialProvider()
        .ifPresent((provider -> this.vendedAdlsCredentialProvider = provider));
```

Nit: using `orElse(null)` isn't really idiomatic of optionals.